### ISSUE TITLE
How billing data is handled

### ISSUE THREAD

**Issue by summer2381** [hystax/optscale#39]
@sd-hystax  May I ask how the bill data is processed, what is the logic of importing, cleaning, and aggregation, and how to operate repetitive data? Is there any relevant documentation? Taking AWS as an example, the first import of bills into mongo raw_expenses, the first How to update and insert in the second time? How to clean and aggregate the original data into the mongo resources and clickhouse expenses tables

**Comment by summer2381** [comment]
hello @maxb-hystax  can you help me to look at the above problem

**Comment by maxb-hystax** [comment]
Hi!
We've not published detailed design documents on solution components yet, but I'll try to describe the flow briefly:
- diworker gets raw expenses data from the cloud (from API or bucket) and stores it in raw_expenses collection. This collection is used for further aggregations
- then, diworker processes data in raw_expenses collection, grouping them into resources (creating them if needed) and aggregates daily expenses by resource and stores it in expenses table in Clickhouse DB, negating previous versions of clean expenses records.
Basically, you can check the code of Diworker service for more details regarding raw expenses processing for different cloud providers - https://github.com/hystax/optscale/tree/integration/diworker/diworker/importers
